### PR TITLE
Library Genesis: Improve Search API, Added Download

### PR DESCRIPTION
**Pull request by yz3440**
## Description

- Deprecated [libgen.js](https://github.com/dunn/libgen.js/) due to poor search results.
- Migrated to a custom scraper that parses raw html from a query request to libgen.
- Changed the default url to open in the browser to the download page for convenience.

## Screencast

screencast of library genesis

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by yz3440** [comment]
I don't know why all the commits before this change is also showing up in the thread. Is there a way to prevent that?
I'm fairly new to contributing to an open-source project, so just trying to learn about the best practices.

**Comment by danielsequeira** [comment]
@yz3440 I just tested the new features of the extension and I found a bug when triggering "Open Book Info Page", it opens a URL starting with "https://libgen.is../book/" and it should be "https://libgen.is/book/". The same happens if I trigger the action to "Copy Book Info URL"

File: extensions/library-genesis/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+- Added two features that are configurable in preferences
+  - Automatically copy text from clipboard to the search field
+  - Set preferred languages and prioritize them in the search results
+
+## [Updated search books] - 2022-07-16
+
+- Deprecated [libgen.js](https://github.com/dunn/libgen.js/) due to poor search results.
+- Migrated to a custom scraper that parse raw html from a query request to libgen.
+- Changed the default url to open in browser to the download page for convenience.
+
```
[the remark this answers is not in the document]
Comment (yz3440) [reply extensions/library-genesis/CHANGELOG.md:19]: Thanks for the suggestion! I changed it to 20 of July for the time being.

File: extensions/library-genesis/package.json [hunk 2]
```diff
@@ -18,12 +18,93 @@
...
+          "title": "Download Book",
+          "value": "2"
+        }
+      ],
+      "default": "",
+      "required": true
+    },
+    {
+      "name": "copySearchContentFromClipboard",
+      "title": "Search field from clipboard",
```
Comment (danielsequeira) [inline extensions/library-genesis/package.json:46]:
```suggestion
      "title": "Search Field from Clipboard",
```

File: extensions/library-genesis/package.json [hunk 3]
```diff
@@ -18,12 +18,93 @@
...
+      "default": "",
+      "placeholder": "English, Spanish, German, ...",
+      "required": false
+    },
+    {
+      "name": "priotizePreferredLanguage",
+      "description": "Prioritizes preferred languages in your searching results.",
+      "type": "checkbox",
+      "default": true,
+      "label": "Prioritize preferred languages in search",
```
Comment (danielsequeira) [inline extensions/library-genesis/package.json:97]:
```suggestion
      "label": "Prioritize Preferred Languages in Search",
```

**Comment by danielsequeira** [comment]
@yz3440 just a few more changes, the fields in the preferences should be title case. After that we can merge 🙂

**Review by danielsequeira — APPROVED** [review]

**Comment by twelvespeedbike** [comment]
I love the extension! However, I can't seem to download any book directly to my computer without going to libgen. When I click the download button (either as a primary action or manually), the window closes and a floating message says "Please select a location to save the book...", but I can't. There's no option to set a default either. Would love to know if I'm doing something wrong or if this is something you're aware of. Thanks!
CleanShot 2022-09-06 at 16 18 29
CleanShot 2022-09-06 at 16 19 09

**Comment by danielsequeira** [comment]
@twelvespeedbike you should report this as a bug using the extension page, this will create an issue and @yz3440 can look into it after.

**Pull request merged**